### ISSUE TITLE
possible bug in OthelloNNet.py?

### ISSUE THREAD

**Issue by jih332** [suragnair/alpha-zero-general#45]
Hi, it's me again. I've noticed that in OthelloNNet.py line 37, you wrote:
`self.prob = tf.nn.softmax(self.pi)`
and in line 48:
`self.loss_pi =  tf.losses.softmax_cross_entropy(self.target_pis, self.prob)`
I just checked the tf doc [softmax_cross_entropy](https://www.tensorflow.org/api_docs/python/tf/losses/softmax_cross_entropy) and it says:
> tf.losses.softmax_cross_entropy(
    onehot_labels,
    logits,
    weights=1.0,
    label_smoothing=0,
    scope=None,
    loss_collection=tf.GraphKeys.LOSSES,
    reduction=Reduction.SUM_BY_NONZERO_WEIGHTS
)

which means the softmaxcross_entropy already implements the softmax for you, and you should provide logits as input to softmax_cross_entropy, not some prob.
Actually, I tried to changed line 48 to
`self.loss_pi =  tf.losses.softmax_cross_entropy(self.target_pis, self.pi)`
and it runs fine as well. 
Please correct me if I got something wrong. Thanks!

**Comment by suragnair** [comment]
It was changed in response to #21 . Looks like that Issue was erroneous, right?

**Comment by jih332** [comment]
yep. using `self.prob` would result in calculating softmax twice. 
Btw, `softmax_cross_entropy_with_logits` is in `tf.nn`, not `tf.losses`. And it's replaced by `softmax_cross_entropy_with_logits_v2` now.

**Comment by suragnair** [comment]
That was extremely careless of me. Thanks for pointing this out. Will fix.
